src/gui/MainWindow.h

```diff
@@ -165,6 +167,9 @@ class MainWindow: public GladeGui, public LayerCtrlListener {
     GtkWidget* winXournal = nullptr;
     ScrollHandling* scrollHandling = nullptr;
 
+    bool usingTouchWorkaround;
+    bool gtkTouchscreenScrollingEnabled;
```

Comment:
According to its description in the settings dialog, the touchscreen workaround should only take effect when the application restarts. 

`usingTouchWorkaround` is cached at program run to provide this functionality -- `settings->isTouchWorkaround()` reflects the current state of the setting. It might make more sense for `settings->isTouchWorkaround()` to return the current **effective state** of `touchWorkaroundEnabled`. Unfortunately, the preferences dialog uses this to set the state of the touch workaround checkbox... 

Similarly, `gtkTouchscreenScrollingEnabled` is necessary as (I think) it isn't stored anywhere else (except GTK).